lib/clerk/authenticate_context.rb

```diff
@@ -0,0 +1,202 @@
+        end
+
+        def clerk_redirect_url
+             # TODO(dimkl): Add multi-domain support
+             ""
+        end
+
+        ##
+        # Helper methods supported only by Ruby SDK to reduce code duplication and improve readability
+        #
+
+        def dev_browser
+            @dev_browser ||= retrieve_from_query_string(@clerk_url, DEV_BROWSER_KEY) || @cookies.dev_browser || ""
+        end
+
+        # The frontend_api returned is without protocol prefix
+        def frontend_api
+            return "" if !publishable_key || !valid_publishable_key?(publishable_key)
+
+            @frontend_api ||= if proxy_url 
+                proxy_url
+            elsif instance_type == "development" && domain
+                "clerk.#{domain}"
+            else
+                # remove $ postfix
+                decode_publishable_key(publishable_key).chop
+            end
+        end
+
+        def instance_type
+            @instance_type ||= publishable_key.start_with?("pk_live_") ? "production" : "development"
+        end
+
+        def development_instance?
+            # TODO(dimkl): Consolidate with @clerk/backend to use pk instead of sk
+            # instance_type == "development"
+
+            secret_key.start_with?("test_") || secret_key.start_with?("sk_test_")
+        end
+
+        def production_instance?
+            # TODO(dimkl): Consolidate with @clerk/backend to use pk instead of sk
+            # instance_type == "production"
+
+            secret_key.start_with?("live_") || secret_key.start_with?("sk_live_")
+        end
+
+        def document_request?
+            @headers.sec_fetch_dest == "document"
+        end
+
+        def accepts_html?
+            @headers.accept && @headers.accept.start_with?('text/html')
+        end
+        
+        def eligible_for_multi_domain?
+            is_satellite? && document_request? && !clerk_synced?
+        end
+
+        def active_client?
+            @cookies.client_uat.to_i > 0
+        end
+
+        def cross_origin_request?
+            # origin contains scheme+host and optionally port (omitted if 80 or 443)
+            # ref. https://www.rfc-editor.org/rfc/rfc6454#section-6.1
+            return false if @headers.origin.nil?
+      
+            # strip scheme
+            origin = @headers.origin.strip.sub(/\A(\w+:)?\/\//, '')
+            return false if origin.empty?
+      
+            # Rack's host and port helpers are reverse-proxy-aware; that
+            # is, they prefer the de-facto X-Forwarded-* headers if they're set
+            request_host = @headers.host
+            request_host << ":#{@headers.port}" if @headers.port != 80 && @headers.port != 443
+      
+            origin != request_host
+        end
+    
+        def browser_request?
+            @headers.user_agent && @headers.user_agent.starts_with?("Mozilla/")
+        end
+
+        def dev_browser?
+            !dev_browser.empty?
+        end
+
+        def header_token?
```

Comment:
This method name is a bit misleading. It feels simpler to have two methods such as `session_token_in_header`, `session_token_in_cookie` and work with them.